File: scripts/trigger-clear-kv.js
#!/usr/bin/env node
/**
 * Trigger wrangler CLI to clear all KV namespaces
 * Can be called from browser via worker endpoint or run directly
 */

const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

// Read wrangler.toml to get all KV namespace IDs
const wranglerPath = path.join(__dirname, '../worker/wrangler.toml');
const wranglerToml = fs.readFileSync(wranglerPath, 'utf-8');

// Parse KV namespaces from wrangler.toml
const namespaces = [];
const kvRegex = /binding\s*=\s*"([^"]+)"[\s\S]*?id\s*=\s*"([^"]+)"/g;
let match;

while ((match = kvRegex.exec(wranglerToml)) !== null) {
  namespaces.push({
    binding: match[1],
    id: match[2]
  });
}

console.log(`🗑️  Clearing ${namespaces.length} KV namespaces...\n`);

const results = [];

for (const ns of namespaces) {
  try {
    console.log(`📦 ${ns.binding} (${ns.id})`);
    
    // List all keys
    const listOutput = execSync(
      `npx wrangler kv:key list --namespace-id=${ns.id}`,
      { encoding: 'utf-8', cwd: path.join(__dirname, '../worker') }
    );
    
    const keys = JSON.parse(listOutput);
    console.log(`   Found ${keys.length} keys`);
    
    if (keys.length === 0) {
      results.push({ namespace: ns.binding, deleted: 0, success: true });
      continue;
    }
    
    // Delete each key
    let deleted = 0;
    for (const key of keys) {
      try {
        execSync(
          `npx wrangler kv:key delete "${key.name}" --namespace-id=${ns.id}`,
          { encoding: 'utf-8', cwd: path.join(__dirname, '../worker'), stdio: 'pipe' }
        );
        deleted++;
      } catch (err) {
        console.error(`   ❌ Failed to delete: ${key.name}`);
      }
    }
    
    console.log(`   ✅ Deleted ${deleted}/${keys.length} keys\n`);
    results.push({ namespace: ns.binding, deleted, total: keys.length, success: true });
    
  } catch (err) {
    console.error(`   ❌ Error: ${err.message}\n`);
    results.push({ namespace: ns.binding, success: false, error: err.message });
  }
}

console.log('\n📊 Summary:');
console.log(JSON.stringify(results, null, 2));

const totalDeleted = results.reduce((sum, r) => sum + (r.deleted || 0), 0);
console.log(`\n✅ Total keys deleted: ${totalDeleted}`);

File: scripts/trigger-clear-kv.py
#!/usr/bin/env python3
"""
Trigger wrangler CLI to clear all KV namespaces
Reads namespaces from wrangler.toml and clears all keys
"""

import subprocess
import json
import re
from pathlib import Path

# Read wrangler.toml to get all KV namespace IDs
wrangler_path = Path(__file__).parent.parent / "worker" / "wrangler.toml"
wrangler_toml = wrangler_path.read_text()

# Parse KV namespaces
namespaces = []
pattern = r'binding\s*=\s*"([^"]+)"[\s\S]*?id\s*=\s*"([^"]+)"'
for match in re.finditer(pattern, wrangler_toml):
    namespaces.append({
        'binding': match.group(1),
        'id': match.group(2)
    })

print(f"🗑️  Clearing {len(namespaces)} KV namespaces...\n")

results = []
worker_dir = Path(__file__).parent.parent / "worker"

for ns in namespaces:
    try:
        print(f"📦 {ns['binding']} ({ns['id']})")
        
        # List all keys
        list_cmd = ['npx', 'wrangler', 'kv:key', 'list', f"--namespace-id={ns['id']}"]
        list_output = subprocess.check_output(list_cmd, cwd=worker_dir, text=True)
        keys = json.loads(list_output)
        
        print(f"   Found {len(keys)} keys")
        
        if len(keys) == 0:
            results.append({'namespace': ns['binding'], 'deleted': 0, 'success': True})
            continue
        
        # Delete each key
        deleted = 0
        for key in keys:
            try:
                delete_cmd = ['npx', 'wrangler', 'kv:key', 'delete', key['name'], f"--namespace-id={ns['id']}"]
                subprocess.run(delete_cmd, cwd=worker_dir, capture_output=True, check=True)
                deleted += 1
            except subprocess.CalledProcessError:
                print(f"   ❌ Failed to delete: {key['name']}")
        
        print(f"   ✅ Deleted {deleted}/{len(keys)} keys\n")
        results.append({
            'namespace': ns['binding'],
            'deleted': deleted,
            'total': len(keys),
            'success': True
        })
        
    except Exception as err:
        print(f"   ❌ Error: {err}\n")
        results.append({'namespace': ns['binding'], 'success': False, 'error': str(err)})

print('\n📊 Summary:')
print(json.dumps(results, indent=2))

total_deleted = sum(r.get('deleted', 0) for r in results)
print(f"\n✅ Total keys deleted: {total_deleted}")
